Dockerfile

```
@@ -38,22 +39,33 @@ RUN \
 
 ARG GH_VERSION=2.72.0
 # ARG GH_SHASUM_256="6df9b0214f352fe62b2998c2d1b9828f09c8e133307c855c20c1924134d3da25"
-RUN curl --silent --show-error --location --output /tmp/gh.tar.gz \
-  "https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}_linux_amd64.tar.gz" \
+RUN ARCH="$(uname -m)" && \
+  if [ "$ARCH" = "x86_64" ]; then \
+    DOWNLOAD_URL="https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}_linux_amd64.tar.gz"; \
+  elif [ "$ARCH" = "aarch64" ]; then \
+    DOWNLOAD_URL="https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}_linux_arm64.tar.gz"; \
+  else \
+    echo "Unsupported architecture: $ARCH" && exit 1; \
+  fi \
+  && curl --silent --show-error --location --output /tmp/gh.tar.gz "${DOWNLOAD_URL}" \
```

Easier to ask `dpkg` on Debian/Ubuntu (when you expect either `amd64` or `arm64` like we do on packer-images (see https://github.com/jenkins-infra/packer-images/blob/68eba541d19885223144cd6d32eca550688e9f7d/provisioning/ubuntu-provision.sh#L194):

```suggestion
RUN curl --silent --show-error --location --output /tmp/gh.tar.gz \
  "https://github.com/cli/cli/releases/download/v${GH_VERSION}/gh_${GH_VERSION}_linux_$(dpkg --print-architecture).tar.gz" \
```

Example of outputs on our productions:

```bash
dduportal@ci.jenkins.io:~$ dpkg --print-architecture
arm64
dduportal@archives:~$ dpkg --print-architecture
amd64
```